[Unified Tabs] Config options and custom button override support

walterra opened the pull request:
## Summary

Fixes #239684.

This PR adds config options to the unified tabs package by introducing feature flag props:

- `enableInlineLabelEditing` - toggle the feature to edit tab labels via double click
- `enablePreview` - toggle the tab preview on hover
- `enableDragAndDrop` - toggle support to drag-n-drop the tab order
- `enableTabsBarMenu` - allows to disable to popover menu to the very right

It also adds support for custom menu buttons on individual tabs (`item?.customMenuButton`) and custom add buttons (`customNewTabButton`). 

This is part of the effort in Lens to use tabs for configuring xy chart layers which doesn't need some of the advanced features of the unified tabs package. The defaults of the new flags have been picked to match the existing behavior in Discover.

The use case for a custom add button (`customNewTabButton `): In Lens this button triggers a popover which gives you some option create different kinds of tabs (data, annotation or reference line tab).

### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

walterra committed:
wip: tab type toggle

walterra committed:
custom create button for unified tabs

walterra committed:
enableInlineLabelEditing flag

walterra committed:
wip: custom unified tabs actions

walterra committed:
customMenuButton

walterra committed:
make drag and drop optional

walterra committed:
add enableTabsBarMenu float

walterra committed:
revert button size to 's'

walterra committed:
refactor: eliminate drag-drop duplication with wrapper components

walterra committed:
refactor: use render props instead of cloneElement

walterra committed:
refactor: extract optional drag wrappers to separate files

walterra committed:
docs: add comments explaining render prop pattern usage

walterra committed:
jest tests

walterra committed:
refactor: simplify tabs drag-drop tests

walterra committed:
tweak comment

walterra committed:
adds jest test for createItemElement

walterra committed:
adds jest test for enableTabsBarMenu

walterra committed:
adds jest test for enableInlineLabelEditing=false

walterra committed:
adds jest test for enablePreview

jughosta commented on src/platform/packages/shared/kbn-unified-tabs/src/components/tab/tab.tsx:
```diff
@@ -252,16 +259,19 @@ export const Tab: React.FC<TabProps> = (props) => {
               </EuiFlexItem>
             )}
-            {!!onClose && (
+            {!!onClose && !item.customMenuButton && (
```
Can we introduce another flag for hiding the Close buttons?

jughosta reviewed (commented):
Great to see that adoption of `unified-tabs` is increasing! Thanks for extending the package!

walterra committed:
refactor unified tabs to use disable instead of enable

walterra committed:
rename createItemElement to customNewTabButton prop

walterra committed:
add disableCloseButton prop with tests to unified tabs

walterra committed:
add tests for customMenuButton in unified tabs

walterra committed:
Merge branch 'main' into 224687-unified-tabs-options

rbrtj committed:
[AI Infra] GenAI Settings: Reflect selected `AI Assistants Visibility` value from header selector on the Settings page (#239555)

Resolves https://github.com/elastic/kibana/issues/236804
The `ai_assistant_management` plugin was using the old
`coreStart.uiSettings` client, while the `GenAI Settings` plugin uses
the newer `coreStart.settings.client`. This mismatch prevented them from
syncing properly.

After:


https://github.com/user-attachments/assets/28e3520e-2f59-4976-b0fa-83e0d32cff3b

umbopepato committed:
[ReponseOps][Connector][Docs] Add JSM connector docs (#237862)

Adds docs for the new JSM connector

---------

Co-authored-by: Nastasha Solomon <nastasha.solomon@elastic.co>
Co-authored-by: florent-leborgne <florent.leborgne@elastic.co>

walterra commented:
@jughosta thanks for the feedback, I addressed all comments, ready for another look.

jughosta commented on src/platform/packages/shared/kbn-unified-tabs/src/components/tabs_bar/tabs_bar.tsx:
```diff
@@ -76,6 +76,8 @@ export type TabsBarProps = Pick<
   onEBTEvent: (event: TabsEBTEvent) => void;
   onClearRecentlyClosed: TabsBarMenuProps['onClearRecentlyClosed'];
+  customNewTabButton?: React.ReactElement;
+  disableTabsBarMenu?: boolean; // defaults to `true`
```
```suggestion
  disableTabsBarMenu?: boolean;
```

jughosta reviewed (approved):
LGTM, thanks!

walterra committed:
update test description and remove incorrect comment

walterra committed:
Merge remote-tracking branch 'upstream/main' into 224687-unified-tabs-options
